src/atoms/TextInput/TextInput.web.js

```diff
@@ -244,8 +262,17 @@ const TextInput = ({
       if (onChange) {
         onChange(text);
       }
+
+      if (_isMultiline && variant === 'outlined') {
+        setTimeout(() => {
+          inputRef.current.style.height = 'inherit';
+          // Calculate the height
+          const height = inputRef.current.scrollHeight;
+          inputRef.current.style.height = `${height}px`;
```

Comment:
Not able to understand what this whole block of code is doing. Please add a comment.
Also, we're setting `inputRef.current.style.height` twice? First `inherit` and then `${height}px`. Why?

Reply:
have moved this to a useEffect. It adjusts the height of textarea based on the input. 
Have added comments.